project/undoable_project_repo.go

```diff
@@ -0,0 +1,84 @@
+package project
+
+import (
+	"golang.org/x/net/context"
+
+	"github.com/almighty/almighty-core/errors"
+	"github.com/almighty/almighty-core/gormsupport"
+	"github.com/jinzhu/gorm"
+	satoriuuid "github.com/satori/go.uuid"
+)
+
+var _ Repository = &UndoableProjectRepository{}
```

Comment:
No need to call it `UndoableProjectRepository` as we are in the `project` namespace. Call it `UndoableRepository`.